### PR TITLE
ORC-644: Support positional mapping for nested types

### PR DESCRIPTION
**Pull request by ArvinZheng**
fix [ORC-644](https://issues.apache.org/jira/browse/ORC-644)

### CODE REVIEW

**Comment by ArvinZheng** [comment]
hi @omalley , mind taking a look at this PR? Thanks.

File: java/core/src/java/org/apache/orc/impl/SchemaEvolution.java [hunk 1]
```diff
@@ -123,7 +123,7 @@ public SchemaEvolution(TypeDescription fileSchema,
           }
         }
       }
-      buildConversion(fileSchema, this.readerSchema, positionalLevels);
+      buildConversion(fileSchema, this.readerSchema, positionalLevels, options.isForcePositionalEvolutionNestedTypes());
```
Comment (omalley) [inline java/core/src/java/org/apache/orc/impl/SchemaEvolution.java:126]: Can't you just set positionalLevels to Integer.MAX_VALUE?

File: java/core/src/java/org/apache/orc/impl/SchemaEvolution.java [hunk 2]
```diff
@@ -18,19 +18,19 @@
 
 package org.apache.orc.impl;
 
+import org.apache.hadoop.conf.Configuration;
```
Comment (omalley) [inline java/core/src/java/org/apache/orc/impl/SchemaEvolution.java:21]: Can you undo the movement of the imports?

File: java/core/src/test/org/apache/orc/impl/TestSchemaEvolution.java [hunk 3]
```diff
@@ -61,6 +44,24 @@
 import org.junit.Test;
 import org.junit.rules.TestName;
 
+import java.io.File;
```
Comment (omalley) [inline java/core/src/test/org/apache/orc/impl/TestSchemaEvolution.java:47]: Please undo the movement of the imports.

**Comment by omalley** [comment]
Actually, doesn't ORC-626 fix this too?

**Comment by ArvinZheng** [comment]
> Actually, doesn't [ORC-626](https://issues.apache.org/jira/browse/ORC-626) fix this too?

Yes, [ORC-626](https://issues.apache.org/jira/browse/ORC-626) fixed this issue, just different ways of implementation, thank you very much for reviewing this.

**Pull request closed without merging**